Library/Homebrew/cleaner.rb

```diff
@@ -153,6 +155,40 @@ def clean_dir(directory)
     end
   end
 
+  sig { void }
+  def rewrite_pkgconfig
+    pc_files = [@formula.lib.realpath, @formula.share.realpath].flat_map do |d|
+      next [] if @formula.skip_clean?(d) || @formula.skip_clean?(d/"pkgconfig")
+
+      d.glob("pkgconfig/*")
+    end
+    return if pc_files.empty?
+
+    keg_only_pc_files = @formula.deps
+                                .reject { |d| d.build? || d.test? }
+                                .map(&:to_formula)
+                                .select(&:keg_only?)
```

Comment:
Perhaps controversial idea: I think this could be usefully done for all formulae, not just keg-only. We've debated for a while about whether we should try to support formulae when `brew unlink`ed and, generally, I think we should.